### PR TITLE
Team 1 combo of previous 2 PR

### PR DESCRIPTION
**Pull request by vsinh14**
https://github.com/UQcsse3200/2024-studio-1/pull/482
https://github.com/UQcsse3200/2024-studio-1/pull/487

### CODE REVIEW

**Review by manya-k — COMMENTED** [review]
crashes at random places due to memory full error but thats probably team 6
upon getting killed by the third boss the dragon death screen comes up

File: source/core/src/main/com/csse3200/game/components/screendisplay/PlayerSelectDisplay.java [hunk 1]
```diff
@@ -52,6 +57,24 @@ public PlayerSelectDisplay(GdxGame game) {
     public void create() {
         super.create();
         addActors();
+
+        // InputMultiplexer for handling both stage and ESC key input
+        InputMultiplexer inputMultiplexer = new InputMultiplexer();
```
Comment (Aryan-S-Gupta) [inline source/core/src/main/com/csse3200/game/components/screendisplay/PlayerSelectDisplay.java:62]: Duplicate Code

File: source/core/src/main/com/csse3200/game/components/settingsmenu/SettingsMenuDisplay.java [hunk 2]
```diff
@@ -71,6 +76,24 @@ private void addActors() {
         rootTable.add(menuBtns).fillX();
 
         stage.addActor(rootTable);
+
+        // InputMultiplexer for handling both stage and ESC key input
```
Comment (Aryan-S-Gupta) [inline source/core/src/main/com/csse3200/game/components/settingsmenu/SettingsMenuDisplay.java:80]: Duplicate Code

**Review by manya-k — APPROVED** [review]

**Review by Aryan-S-Gupta — COMMENTED** [review]
When the player gets killed by boss, the dragon is shown on the deathscreen at times

**Review by Aryan-S-Gupta — APPROVED** [review]

**Pull request merged**